engines/wintermute/ad/ad_scene_geometry.cpp

```diff
@@ -496,74 +496,49 @@ bool AdSceneGeometry::render(bool render) {
-	//	m_Renderer->SetSpriteBlendMode(BLEND_UNKNOWN);
-	//	m_Renderer->m_Device->SetRenderState(D3DRS_SRCBLEND,  D3DBLEND_ZERO);
-	//	m_Renderer->m_Device->SetRenderState(D3DRS_DESTBLEND, D3DBLEND_ONE);
+	glFrontFace(GL_CW);
+	glBindTexture(GL_TEXTURE_2D, 0);
 
-	//	m_Renderer->m_Device->SetTransform(D3DTS_WORLD, &matIdentity);
-	//	C3DUtils::SetFixedVertexShader(m_Renderer->m_Device, D3DFVF_MODELVERTEXCOLOR);
+	// render walk planes
+	for (uint i = 0; i < _planes.size(); i++) {
+		if (!_planes[i]->_active || !_planes[i]->_receiveShadows) {
+			continue;
+		}
 
-	//	// no texture
-	//	m_Renderer->m_LastTexture = NULL;
-	//	m_Renderer->m_Device->SetTexture(0, NULL);
+		_planes[i]->_mesh->render();
 
-	//	m_Renderer->m_Device->SetRenderState(D3DRS_CULLMODE, D3DCULL_CCW);
+		//m_Renderer->m_NumPolygons += _planes[i]->m_Mesh->m_NumFaces;
+	}
 
-	//	// render blocks
-	//	for(int i=0; i<_blocks.size(); i++)
-	//	{
-	//		if(!_blocks[i]->m_Active) continue;
-	//		if(!_blocks[i]->m_Mesh->m_VB) continue;
-	//		if(!_blocks[i]->m_ReceiveShadows) continue;
-	//		C3DUtils::SetStreamSource(m_Renderer->m_Device, 0, _blocks[i]->m_Mesh->m_VB, sizeof(MODELVERTEXCOLOR));
-	//		m_Renderer->m_Device->DrawPrimitive(D3DPT_TRIANGLELIST, 0, _blocks[i]->m_Mesh->m_NumFaces);
+	// render blocks
+	for (uint i = 0; i < _blocks.size(); i++) {
+		if (!_blocks[i]->_active || !_blocks[i]->_receiveShadows) {
+			continue;
+		}
 
-	//		m_Renderer->m_NumPolygons += _blocks[i]->m_Mesh->m_NumFaces;
-	//	}
+		_blocks[i]->_mesh->render();
 
-	//	// render walkplanes
-	//	for(int i=0; i<_planes.size(); i++)
-	//	{
-	//		if(!_planes[i]->m_Active) continue;
-	//		if(!_planes[i]->m_Mesh->m_VB) continue;
-	//		if(!_planes[i]->m_ReceiveShadows) continue;
-	//		C3DUtils::SetStreamSource(m_Renderer->m_Device, 0, _planes[i]->m_Mesh->m_VB, sizeof(MODELVERTEXCOLOR));
-	//		m_Renderer->m_Device->DrawPrimitive(D3DPT_TRIANGLELIST, 0, _planes[i]->m_Mesh->m_NumFaces);
+		//		m_Renderer->m_NumPolygons += _blocks[i]->m_Mesh->m_NumFaces;
+	}
 
-	//		m_Renderer->m_NumPolygons += _planes[i]->m_Mesh->m_NumFaces;
-	//	}
+	// render generic objects
+	for (uint i = 0; i < _generics.size(); i++) {
+		if (!_generics[i]->_active || !_generics[i]->_receiveShadows) {
+			continue;
+		}
 
-	//	// render generic meshes
-	//	for(int i=0; i<_generics.size(); i++)
-	//	{
-	//		if(!_generics[i]->m_Active) continue;
-	//		if(!_generics[i]->m_Mesh->m_VB) continue;
-	//		if(!_generics[i]->m_ReceiveShadows) continue;
-	//		C3DUtils::SetStreamSource(m_Renderer->m_Device, 0, _generics[i]->m_Mesh->m_VB, sizeof(MODELVERTEXCOLOR));
-	//		m_Renderer->m_Device->DrawPrimitive(D3DPT_TRIANGLELIST, 0, _generics[i]->m_Mesh->m_NumFaces);
+		_generics[i]->_mesh->render();
 
-	//		m_Renderer->m_NumPolygons += _generics[i]->m_Mesh->m_NumFaces;
-	//	}
+		//		m_Renderer->m_NumPolygons += _generics[i]->m_Mesh->m_NumFaces;
+	}
 
-	//	m_Renderer->SetSpriteBlendMode(BLEND_NORMAL);
+	_gameRef->_renderer3D->setSpriteBlendMode(Graphics::BLEND_NORMAL);
```

Comment:
Does any of the other rendering depend on this state? If so, perhaps it's better to query the preceeding state, to ensure that you set the precise state back.

Reply:
All the other rendering code will set it's own state for this as far as I know anyways. Wme also just sets it like this.